specs/PushNotifications/PushNotifications-spec.md

```diff
@@ -184,21 +194,34 @@ int main()
         }
     }
 
-    // Unregisters the inproc COM Activator before exiting
-    PushNotificationManager::UnregisterActivator(token, PushNotificationRegistrationKind::ComActivator);
+    if(isComActivationSupported)
+    {
+        // Unregister the ComActivator from inproc activation
+        PushNotificationManager::UnregisterActivator(token, PushNotificationRegistrationOptions::ComActivator);
```

Comment:
Maybe this could be `PushNotificationManager::UnregisterActivator(token)` because the registration step remembers whether it was registered for Com or launch or...

Reply:
We have to let developer decide which flag they want to unregister. They should call UnregisterActivator with PushTrigger if they no longer want to receive PushNotifications. They should call UnregisterActivator with ComActivator to unregister the calling app from inproc activation.

A common pattern will be RegisterActivator(ComActivator | PushTrigger) -> UnregisterActivator(ComActivator).

Reply:
I see. I'd assumed that a developer would register twice, once per activation kind, rather than or-ing together flags, which meant that they got unique tokens back from registration.

Is there an "unregister all my activators" in case the app can't remember the tokens it got?

Reply:
We do not have an "unregister all my activators" function. Should we consider adding an API like this?

Reply:
I'd think so. Handy for apps that do conditional activations (varying options) can have 1 cleanup call `UnregisterActivatorAllKinds()` and not have to track which options they original passed in

Reply:
We can store the RegistrationTokens for the user and revoke all tokens when they call this function. Should RegisterActivator then not return a token and we just handle it for the user? And UnregisterActivator just accept flags? (What @jonwis said above)
`void RegisterActivator(PushNotificationActivationInfo details);`

`void UnregisterActivator(PushNotificationRegistrationActivators options);